### ISSUE TITLE
Importing Extras Independently

### ISSUE THREAD

**Issue by craycrayfish** [aineko-dev/aineko#122]
All extras need to be installed for a single extra to be used because they are imported in `extras/__init__.py`. We should make them independent by adding moving the import down a level, such as `extras/fastapi/__init__.py`.

**Comment by Barabazs** [comment]
How about doing something like this in `extras/__init__.py`

```python
try:
    from aineko.extras.fastapi.main import FastAPI
except ImportError:
    pass
```

**Comment by craycrayfish** [comment]
I think it's better not to do the try-except, because that can mask other issues

**Comment by Barabazs** [comment]
I did some research and the proper way of doing it would be trough plugins. [source](https://packaging.python.org/en/latest/guides/creating-and-discovering-plugins/#creating-and-discovering-plugins)

**TL;DR:**
* the way we're using extras is meant for optional dependencies, not for optional "features" (I'll use plugin instead of extra from now on)
* each plugin needs to be packaged and distributed separately
* we can have a common namespace like aineko_plugins for imports (e.g. `from aineko_plugins import websocket_client`)
* users would still be able to install plugins together with the main aineko package if they want. 
* we can still have a monorepo for aineko and the plugins

**Production example:** 
Apache Airflow ships each provider integration as a separate plugin.
All providers live under the same project and monorepo at https://github.com/apache/airflow/tree/main/airflow/providers/
For example the google provider is published here: https://pypi.org/project/apache-airflow-providers-google/
And you can either install it independently with `pip install apache-airflow-providers-google` or together with the main package with `pip install apache-airflow[google]`

**Comment by Barabazs** [comment]
Resolved with https://github.com/aineko-dev/aineko/pull/154 and https://github.com/aineko-dev/aineko-plugins
